java/plugins/web-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/WebElementsPage.java

```diff
@@ -0,0 +1,497 @@
+     * catch it. Text is being waited by {@link #assertTextAppears}, so it will
+     * be space-trimmed as well
+     *
+     * @param text text that will be searched inside of the window
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.WaitException if
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.modal.window.with.text.appears")
+    public void assertModalWindowAppears(String text) throws WaitException {
+        try {
+            String popupHandle = WebExtension.findNewWindowHandle((Set<String>) Stash.getValue("beforeClickHandles"));
+            if (null != popupHandle && !popupHandle.isEmpty()) {
+                PageFactory.getWebDriver().switchTo().window(popupHandle);
+            }
+            assertTextAppears(text);
+        } catch (Exception ex) {
+            throw new WaitException("Modal window with text '" + text + "' didn't appear during timeout", ex);
+        }
+    }
+
+    /**
+     * Perform {@link #checkValue(String, WebElement, MatchStrategy)} for the
+     * WebElement with corresponding title on a current page. Use exact match
+     * strategy
+     *
+     * @param text string value that will be searched inside of the element
+     * @param elementTitle title of the element to search
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.ElementNotFoundException if
+     * couldn't find element by given title, or current page isn't initialized
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.check.value")
+    public void checkValue(String elementTitle, String text) throws PageException {
+        checkValue(text, PageFactoryUtils.getElementByTitle(PageContext.getCurrentPage(), elementTitle), MatchStrategy.EXACT);
+    }
+
+    /**
+     * Perform {@link #checkValue(String, WebElement, MatchStrategy)} for the
+     * specified WebElement. Use exact match strategy
+     *
+     * @param text string value that will be searched inside of the element
+     * @param webElement WebElement to check
+     */
+    public void checkValue(String text, WebElement webElement) {
+        checkValue(text, webElement, MatchStrategy.EXACT);
+    }
+
+    /**
+     * Define a type of given WebElement, and check whether it either contains,
+     * or exactly matches given text in its value. Currently supported elements
+     * are text input and select box TODO: use HtmlElements here, to define
+     * which element we are dealing with
+     *
+     * @param text string value that will be searched inside of the element
+     * @param webElement WebElement to check
+     * @param searchStrategy match strategy. See available strategies in
+     * {@link MatchStrategy}
+     */
+    protected void checkValue(String text, WebElement webElement, MatchStrategy searchStrategy) {
+        String value = "";
+        switch (searchStrategy) {
+            case EXACT:
+                try {
+                    switch (webElement.getTagName()) {
+                        case "input":
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "input" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)